schema/index.ts

```diff
@@ -56,6 +60,35 @@ const TransactionResult = objectType({
   },
 });
 
+async function getTranscation(__network: string, hash: string, loop: boolean) {
+  const UnCoverEndpoint =
+    __network == "CENNZnet_Nikau"
+      ? "https://service.eks.centrality.me/"
+      : "https://service.eks.centralityapp.com/";
+
+  while (loop) {
+    // TODO: this may never end, yikes.
```

Comment:
yeah this is a part we should talk about, currently im using uncover as the backend, but it doesn't really make sense, we should run our own archive node to get information that you can't get by querying the node. Im think we should use https://github.com/paritytech/substrate-archive so we arnt rely on uncover as part of the acrtecture and can do more complex SQL queries

Reply:
the reason for this loop, is because I don't know when uncover is going to pickup the transaction, sometimes its not there straight away, but I know it will be there at some point cause I have comfermation that the transaction is on the blockchain. So this loop just keeps trying until its there.

The reason loop is optional is because when you do the mutation it first checks to see if that transaction is already there, if its not then it just returns the base case.

trans = getTransaction from uncover
if trans exists
return trans
else 
submit transaction
 wait for conformation from the CENNZnet node
Loop (trans = getTransaction from uncover) until its loaded into uncover and doesn't return Unknown.
return trans.